magma/passes/passes.py

```diff
@@ -39,6 +39,30 @@ def run(self):
         return self
 
 
+class CircuitPass(Pass):
```

Comment:
Since we're making serious use of the passes, let's take this opportunity to clean up these designs.

`CircuitPass` seems like a good idea, but `DefinitionPass` can probably inherit from it:

```python
class DefinitionPass(CircuitPass):
    def _run(self, circuit):
        if not isdefinition(circuit):
            return
        super()._run(circuit)
```
(actually currently in `DefinitionPass` there is a redundant `isdefinition()` check on instance types).